gcloud/storage/bucket.py

```diff
@@ -205,6 +205,11 @@ def path(self):
 
         return self.path_helper(self.name)
 
+    @property
```

Comment:
We can move it later, but having methods use `self.client` when the ctor only assigns `self._client` and the property definintion is buried hundreds of lines down is a readability issue.